calculator/src/main/java/com/market/banica/calculator/service/ProductServiceImpl.java

```diff
@@ -168,27 +167,29 @@ public void deleteProductFromParentIngredients(String parentProductName, String
     }
 
     @Override
-    public List<Product> getProductAsListProduct(String productName) {
-        LOGGER.debug("In getProductAsListProduct method with parameters:productName {}"
+    public Map<Product, Map<String,Long>> getProductIngredientsWithQuantityPerParent(String productName) {
+        LOGGER.debug("In getProductIngredientsWithQuantity method with parameters:productName {}"
                 , productName);
 
         Product product = getProductFromDatabase(productName);
 
-        List<Product> result = new ArrayList<>();
-
-        result.add(product);
+        Map<Product, Map<String,Long>> result = new HashMap<>();
 
         if (!product.getIngredients().isEmpty()) {
-            addAllIngredientsFromProductInListAsProduct(result, product);
+            addAllIngredientsFromProductToMapOfProductAndQuantity(result, product);
         }
 
-        LOGGER.debug("GetProductAsListProduct with product name {} successfully invoked", productName);
+        LOGGER.debug("GetProductIngredientsWithQuantity with product name {} successfully invoked", productName);
         return result;
     }
 
-    //TODO to be implemented once expectations are clear
     @Override
-    public void getAllProductsAsListProduct() {
+    public Product getProductFromDatabase(String productName) {
+        LOGGER.debug("In getProductFromDatabase method");
+
+        validateProductExists(productName);
```

Comment:
keyword is missing for calling the method

Reply:
Fixed